backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_OrderLine.java

```diff
@@ -126,6 +141,42 @@ public void setQtyEnteredInPriceUOM(final I_C_OrderLine orderLine)
 		orderLine.setQtyEnteredInPriceUOM(qtyEnteredInPriceUOM);
 	}
 
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW,
+			ModelValidator.TYPE_BEFORE_CHANGE
+	}, ifColumnsChanged = {
+			I_C_OrderLine.COLUMNNAME_M_Product_ID
+	})
+	public void checkHaddexDate(final I_C_OrderLine orderLine)
+	{
+		if (!orderDAO.getById(OrderId.ofRepoId(orderLine.getC_Order_ID())).isSOTrx())
+		{
+			return;
+		}
+
+		final I_M_Product product = productDAO.getById(orderLine.getM_Product_ID());
+
+		if (product.isHaddexCheck() && product.getDateHaddexCheck() != null)
+		{
+			final long differenceBetweenHaddexCheckDateAndPromisedDateInMonths = Math.abs(
+					ChronoUnit.MONTHS.between(
+							TimeUtil.asZonedDateTime(product.getDateHaddexCheck()),
+							TimeUtil.asZonedDateTime(orderLine.getDatePromised())
```

Comment:
I'm not 100% sure that the order line already has the right DatePromised on beforeNew or beforeChange.
Therefor i think it's better if you take the DatePromised valud from the C_Order which you load in line 151

Reply:
ok. It makes sense.